src/test/java/ch/wisv/areafiftylan/EventRestIntegrationTest.java

```diff
@@ -0,0 +1,769 @@
+
+        team1 = teamRepository.save(new Team("team1", teamCaptain));
+        team2 = new Team("team2", teamCaptain);
+        team2.addMember(user);
+        team2 = teamRepository.save(team2);
+    }
+
+    @After
+    public void cleanupEventTests() {
+        eventRepository.deleteAll();
+        teamRepository.deleteAll();
+    }
+
+    //region Test Add Events
+
+    @Test
+    public void testAddEventAsAnon() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamLimit", "10");
+        eventDTO.put("teamSize", "2");
+        //@formatter:off
+        given().
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_FORBIDDEN);
+        //@formatter:on
+    }
+
+    @Test
+    public void testAddEventAsUser() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamLimit", "10");
+        eventDTO.put("teamSize", "2");
+
+        SessionData login = login("user");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_FORBIDDEN);
+        //@formatter:on
+    }
+
+    @Test
+    public void testAddEventAsAdmin() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamLimit", "10");
+        eventDTO.put("teamSize", "2");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_CREATED).
+            header("location", containsString("/events/")).
+            body("object.eventName", is("event"));
+        //@formatter:on
+    }
+
+    private void addEvent(String username, String name, String teamLimit, String teamSize) {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", name);
+        eventDTO.put("teamLimit", teamLimit);
+        eventDTO.put("teamSize", teamSize);
+
+        SessionData login = login(username);
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_CREATED).
+            header("location", containsString("/events/")).
+            body("object.eventName", is("event"));
+        //@formatter:on
```

Comment:
I'd also logout in this method. This makes sure the global state stays the same upon calling this code (except for the known addition of an event.) Else, this should be named addEventAndLogin.